src/app/logic/petition/sign-petition.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@angular/core';
+import {
+  exhaustMap,
+  map,
+  merge,
+  Observable,
+  partition,
+  shareReplay,
+  Subject,
+  tap,
+} from 'rxjs';
+import { Result } from 'src/app/shared/models/exports';
+import { SignaturePetitionData } from 'src/app/shared/models/petition/signature-petition-data';
+import { PetitionService } from './petition.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SignPetitionService {
+  public error$: Observable<Result<string>>;
+  public success$: Observable<Result<string>>;
+  public loading$: Observable<boolean>;
+  public result$: Observable<Result<string>>;
+  private submit$: Subject<SignaturePetitionData> = new Subject();
+
+  constructor(private _petitionService: PetitionService) {
+    this.result$ = this.submit$.pipe(
+      exhaustMap((data) => this._petitionService.signaturePetition(data)),
+      shareReplay(1)
+    );
+    const [success$, error$] = partition(this.result$, (value) =>
+      value.result ? true : false
+    );
+
+    this.success$ = success$.pipe(
+      tap((value) => console.log(value)),
+      shareReplay(1)
+    );
+
+    this.error$ = error$.pipe(
+      tap((value) => console.log(value)),
+      shareReplay(1)
+    );
+
+    const end$ = merge(this.success$, this.error$);
+
+    this.loading$ = merge(
+      this.submit$.pipe(
+        map((v) => true),
+        tap(() => console.log('start'))
+      ),
+      end$.pipe(
+        map((v) => false),
+        tap(() => console.log('end'))
+      )
+    ).pipe(shareReplay(1));
+  }
+  ngOnDestroy(): void {
+    this.submit$.complete();
+  }
+
+  set formGroupValue(value: SignaturePetitionData) {
```

Comment:
This should not be a setter, is hard to figure out that this is triggering an action and consequently a backend request. Use a method instead. Ex. `signPetition(data: SignaturePetitionData): void {...}`